core/src/main/scala/sbt/librarymanagement/ResolverExtra.scala

```diff
@@ -397,8 +397,16 @@ private[librarymanagement] abstract class ResolverFunctions {
       new File(sbt.io.Path.userHome, ".m2/repository")
   }
   // TODO - should this just be the *exact* same as mavenLocal?  probably...
-  def publishMavenLocal: MavenCache = new MavenCache("publish-m2-local", mavenLocalDir)
-  def mavenLocal: MavenRepository = new MavenCache("Maven2 Local", mavenLocalDir)
+  def publishMavenLocal: MavenCache =
+    new MavenCache(
+      "publish-m2-local",
+      Option(System.getProperty("maven.repo.local")).map(new File(_)) getOrElse mavenLocalDir
```

Comment:
I'm generally fine with the idea of respecting the system property used by Maven, but I feel like this logic should go into `mavenLocalDir` just above it that goes through multiple stages like looking for `M2_HOME` environment variable to figure out the Maven local location.

Reply:
Agreed, somehow I was under the impression that `mavenLocalDir` was supposed to be the maven home directory.